### DISCUSSION TITLE
Modelling evapotranspiration

### DISCUSSION THREAD

**Discussion by ychopard** [phydrus/phydrus#28] [Q&A] [answered]
Hi
Is it possible to add climate data to the model to calculate evapotranspiration? (Like in the original Hydrus1D version)
Best
Yves

**Comment by mvremec** [comment, marked as answer]
Hi ychopard!
Adding climate data to the model to calculate evapotranspiration is possible but is not yet supported in Phydrus (there is room for improvements here and you are welcome to collaborate). 
For now, you can use external libraries to estimate evapotranspiration and then use the Ritchie (1972) formula to calculate evaporation (rSoil) and transpiration (rRoot) separately:
rRoot = ET * (1 - e**(-k*LAI))
rSoil = ET * e**(-k*LAI)

There are a couple of Python libraries you can use:
- PyETo (works great with Numpy arrays)
https://github.com/mvremec/PyETo
- PyEt (works with Pandas Series)
https://github.com/mvremec/PyEt
Best,
Matevz

**Reply by LIZAO-coder** [reply]
Thank you for your response. If the input is the actual evaporation and transpiration, how should the parameters be set?

**Comment by ychopard** [comment]
Thanks for your answer!

**Comment by raoulcollenteur** [comment]
@ychopard you can actually find PyEt under https://github.com/phydrus/pyet now!

**Comment by ychopard** [comment]
I hesitate to calculate PET outside of Hydrus. That's why I'm still using the old GUI altough I would prefer using phydrus.
I fear mixing different concepts and parameters when I calculate ETc using PyET and split it up using LAI, e.g this is discussed here: https://www.pc-progress.com/forum/viewtopic.php?f=4&t=1015
Do you maybe have any insight on that issue?

**Comment by mvremec** [comment]
Thanks for the question!
For now, I can not state which of the methods, presented in the discussion is best to estimate and split ETc, but what the discussion shows is that there are other methods that have to be taken into consideration, but are unable, as they are not implemented in the GUI.
And exactly this is the purpose of PyEt, where you can either use the same procedure as in the GUI
or use other ETc estimation and split methods to obtain better results.
PyEt method `pm_fao1990` follows the same procedure of estimating ETc as implemented in the GUI, therefore you should obtain the same results as with the GUI.
I will try to come up with a test(notebook) in the next days, which will show that PyEt can reproduce the GUI estimated evaporation and transpiration (probably example 4 from the tutorials).

**Reply by fidaulmustafa** [reply]
Hi mvremec,

Thats great. Did you come up with the test(notebook)?
That could be really helpful.
